llmfoundry/utils/warnings.py

```diff
@@ -51,7 +51,7 @@ def __init__(self, feature_name: str) -> None:
 
 
 # Decorator version of experimental warning
-def experimental(feature_name: str) -> Callable[[F], F]:
+def experimental_function(feature_name: str) -> Callable[[F], F]:
```

Comment:
Can we combine to still be a single function with a simple name `experimental`?
```
def experimental(feature_name: str) -> Callable[[F], F]:
    """Class decorator to mark a class as experimental."""
    def decorator(target: Union[Type, Callable]):
         if isinstance(target, type):
              return decorate_class(target)
        elif callable(target):
              return decorate_fn(target)
        raise Exception()
     return decorator
```

Reply:
I find this a little bit less clear with the typing so I am going to leave it as is